### PR TITLE
[Google Blockly] Add script to compute new accessible colors

### PR DESCRIPTION
**Pull request by mikeharv**
Alternative to:
- https://github.com/code-dot-org/code-dot-org/pull/50430

Please review the above pull request for context. If you are already familiar with the code changes there, you will primarily want to review this new commit: [91d3406](https://github.com/code-dot-org/code-dot-org/pull/50504/commits/91d3406a070b60c267570c0fb9d2cf7708b71f85)

During code review, @fisher-alice expressed hesitations about needing to run a script to generate accessible styles on every page load. @molly-moen pointed out that our pages are already slow-loading and shared the same hesitance. [[Slack thread](https://codedotorg.slack.com/archives/C03DBDN67B7/p1677534394061369)]

## Proposal

As we do not expect to update our default styles all that often, we can instead make this script available to engineers to use when needed. The script and actual `cdoBlockStyles` object were both moved to module files to better facilitate running the script via node in terminal. Comments have been left in each updated file to explain how to update the accessible theme.

**Sample Terminal output** (After adding `nearest-color` package):
image

It is hoped that this proposal will provide a nice middle ground where our team can easily maintain our Blockly themes with causing impacting performance for our users.

After pulling these changes, you can locally add `nearest-color` and run the script to confirm we get valid styles as output. If you make changes to `cdoBlockStyles` first, you should see a diff once you paste in the new accessible colors.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/themes/accessibleColors.mjs [hunk 1]
```diff
@@ -58,12 +63,22 @@ var accessiblePalette = {
...
+  const nearestAvailableColor = nearestColor.from(accessiblePalette);
+  value.colourPrimary = nearestAvailableColor(value.colourPrimary).value;
   // Remove the color from the available palette so we don't assign it twice.
   delete accessiblePalette['color_' + value.colourPrimary.slice(1)];
 }
 
-export default blockStyles;
+console.log(blockStyles);
+console.log(
+  '\x1b[33m%s\x1b[0m',
```
Comment (mikeharv) [inline apps/src/blockly/themes/accessibleColors.mjs:77]: This just tells terminal to use yellow text to better stand out against what is being copied.
https://stackoverflow.com/questions/9781218/how-to-change-node-jss-console-font-color

File: apps/src/blockly/themes/accessibleColors.mjs [hunk 2]
```diff
@@ -58,12 +63,22 @@ var accessiblePalette = {
   color_FF4235: '#FF4235'
 };
 
+console.log(
+  '\x1b[33m%s\x1b[0m',
```
Comment (mikeharv) [inline apps/src/blockly/themes/accessibleColors.mjs:63]: This just tells terminal to use yellow text to better stand out against what is being copied.
https://stackoverflow.com/questions/9781218/how-to-change-node-jss-console-font-color

File: apps/src/blockly/themes/cdoBlockStyles.mjs [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Whenever updating the styles in this file, be sure to also update the styles
```
Comment (fisher-alice) [inline apps/src/blockly/themes/cdoBlockStyles.mjs]: Maybe add an IMPORTANT! at the beginning of this comment.

File: apps/src/blockly/themes/cdoBlockStyles.mjs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+// in cdoAccessible.js as well. The accessible styles can be generated using
+// accessibleColors.mjs.
+// From this directory in your terminal, enter:
```
Comment (fisher-alice) [inline apps/src/blockly/themes/cdoBlockStyles.mjs:4]: Is there a reason you have these instructions in both files?

File: apps/src/blockly/themes/cdoAccessible.js [hunk 5]
```diff
@@ -1,97 +1,38 @@
...
-export const cdoAccessibleBlockStyles = {
-  ...coreBlocklyOverrides,
-  ...cdoCustomAccessibleStyles
+// Whenever updating ./cdoBlockStyles.mjs, be sure to also update the styles
+// here as well. The accessible styles can be generated using accessibleColors.mjs.
+// From this directory in your terminal, enter:
+// - yarn add nearest-color
+// - node accessibleColors.mjs
+// - yarn remove nearest-color
+// Copy the values output by the script into cdoAccessible and commit both changes together.
```
Comment (fisher-alice) [inline apps/src/blockly/themes/cdoAccessible.js]: You could reiterate explicitly what 'both changes' are here.

**Review by fisher-alice — APPROVED** [review]
This is super cool! Thanks for being flexible about modifying your initial approach to updating these style colors.
Just a couple nit comments - great contribution!

File: apps/src/blockly/themes/cdoBlockStyles.mjs (hunk 4 above)
Comment (mikeharv) [reply apps/src/blockly/themes/cdoBlockStyles.mjs:4]: If someone is updating the default styles, I want them to know to update the accessible ones too. If someone happens to be checking out the accessible styles only, I want them to understand where they should come from.

Comment (fisher-alice) [reply apps/src/blockly/themes/cdoBlockStyles.mjs:4]: That makes sense  - thanks!

**Pull request merged**